common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOtherCache.java

```diff
@@ -0,0 +1,68 @@
+package com.microsoft.identity.common.internal.cache;
+
+import android.content.Context;
+import androidx.annotation.VisibleForTesting;
+import com.microsoft.identity.common.adal.internal.cache.IStorageHelper;
+import com.microsoft.identity.common.adal.internal.cache.StorageHelper;
+
+public class MsalOtherCache {
+
+    /**
+     * The name of the SharedPreferences file on disk.
+     * This file is only used to store MSAL CPP authority validation metadata
+     */
+    public static final String DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES =
+            "com.microsoft.identity.client.cpp_authority_validation_metadata";
+
+    private final ICacheKeyValueDelegate mCacheKeyValueDelegate;
+
+    private final ISharedPreferencesFileManager mCppAuthorityValidationMetadataSharedPreferencesFileManager;
+
+    public MsalOtherCache(final Context context)
+    {
+        mCacheKeyValueDelegate = new CacheKeyValueDelegate();
+
+        final IStorageHelper storageHelper = new StorageHelper(context);

```

Comment:
This is going to cause this file to be encrypted.... I don't think you probably want this behavior. I would suggest just passing `null`

Reply:
I think encryption is needed, since the authority in the cache means **verified authority**, and it is possible that hackers may write a fake authority in the cache to steal credentials. I will also discuss this with my colleagues.

Reply:
I have changed the cache file to unencrypted, thanks.